---
layout: post2
permalink: /pathway_and_network_analysis_of_omics_data_2016/
title: Pathway and Network Analysis of -Omics Data 2016 Student Page
header1: Pathway and Network Analysis of -Omics Data 2016
header2: Workshop pages for students
image: CBW_pathway_icon.jpg
---

<ul id="navmenu">
  <li><a id="back_to_top">Contents</a>
     <ul class="sub1">
     <li><a href="#course_schedule">Course Schedule</a></li>
     <li><a href="#q_a_forum">Workshop Q/A Forum</a></li>
     <li><a href="#laptop_setup">Laptop Setup Instructions</a></li>
     <li><a href="#pre_tutorials">Pre-Workshop Tutorials</a></li>
     <li><a href="#pre_readings">Pre-Workshop Readings</a></li>
      <li><a href="#day1">Day 1</a>
         <ul class="sub2">  
           <li><a href="#welcome">Welcome</a></li>
           <li><a href="#module_1">Module 1</a></li>
           <li><a href="#module_2">Module 2</a></li>
           <li><a href="#module_3">Module 3</a></li>
           <li><a href="#assignment1">Integrated Assignment</a></li>
        </ul>
      </li>
       <li><a href="#day_2">Day 2</a>
          <ul class="sub2">
             <li><a href="#module_4">Module 4</a></li>
             <li><a href="#module_5">Module 5</a></li>
             <li><a href="#assignment2">Integrated Assignment</a></li>
           </ul>
       </li>
       <li><a href="#day_3">Day 3</a>
          <ul class="sub2">
             <li><a href="#module_6">Module 6</a></li>
           </ul>
       </li>
    </ul>
  </li>
</ul>  

<br>

###  Course Schedule  <a id="course_schedule"></a>

  <a href="http://bioinformatics-ca.github.io/2016_workshops/pathways/Pathways_2016_Schedule_v3.pdf">Schedule for June 13 to June 15, 2016</a>

[&uarr;](#back_to_top)

###  Workshop Q/A Forum <a id="q_a_forum"></a>

  Post your workshop questions <a href="https://noteapp.com/Pathways2016">here</a>!
  
### Class Photo

![Class photo](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/blob/master/2016_workshops/pathways/img/pathways_class_pic.jpeg?raw=true)

[&uarr;](#back_to_top)

###  Laptop Setup Instructions <a id="laptop_setup"></a>

  Instructions to setup your laptop can be found <a href="http://bioinformatics-ca.github.io/2016_workshops/pathways/laptop_setup_instructions.pdf">here</a>.

[&uarr;](#back_to_top)

###  Pre-workshop Tutorials <a id="pre_tutorials"></a>

1) **Cytoscape Preparation tutorials**: Complete the [introductory tutorial to Cytoscape](http://opentutorials.cgl.ucsf.edu/index.php/Portal:Cytoscape3)

* Introduction to Cytoscape3 - User Interface

* Introduction to Cytoscape3 - Welcome Screen

* Filtering and Editing in Cytoscape 3

[&uarr;](#back_to_top)

###  Pre-workshop Readings <a id="pre_readings"></a>

  Before coming to the workshop, read these.
  
  * [MEME-ChIP: motif analysis of large DNA datasets](http://www.ncbi.nlm.nih.gov/pubmed/21486936)
  
  * [The GeneMANIA prediction server: biological network integration for gene prioritization and predicting gene function](http://www.ncbi.nlm.nih.gov/pubmed/20576703)
  
  * [GeneMANIA Prediction Server 2013 Update](http://www.ncbi.nlm.nih.gov/pubmed/23794635)
  
  * [How to visually interpret biological data using networks](http://www.ncbi.nlm.nih.gov/pubmed/19816451)
  
  * [g:Profiler--a web-based toolset for functional profiling of gene lists from large-scale experiments](http://www.ncbi.nlm.nih.gov/pubmed/17478515)
  
  * [g:Profiler--a web server for functional interpretation of gene lists (2011 update)](http://www.ncbi.nlm.nih.gov/pubmed/21646343)
  
  * [Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles](http://www.ncbi.nlm.nih.gov/pubmed/16199517)
  
  * [Expression data analysis with reactome](http://www.ncbi.nlm.nih.gov/pubmed/25754994)

[&uarr;](#back_to_top)  

***

##  Day 1 <a id="day_1"></a>

###  Welcome <a id="welcome"></a>

  *<font color="#827e9c">Ann Meyer</font>* 
<br>

[&uarr;](#back_to_top)

***

###  Module 1: Introduction to Pathway and Network Analysis <a id="module_1"></a>

  *<font color="#827e9c">Gary Bader</font>*
  
  [Lecture](https://bioinformatics.ca/pathways-module-1)

[&uarr;](#back_to_top)

***

###  Module 2: Finding Over-Represented Pathways in Gene Lists <a id="module_2"></a>

  *<font color="#827e9c">Quaid Morris</font>*
  
[Lecture](https://bioinformatics.ca/pathways-module-2)
  
#### Lab practical: Enrichment-Based Analysis - Performing ORA
  *<font color="#827e9c">Veronique Voisin</font>*
  
  [Introduction](https://bioinformatics.ca/pathways-module-2-lab-2016) 
  
  [Exercise 1](http://bioinformatics-ca.github.io/PNAOD_module2_lab_GSEA_2016/)
  
  [Precomputed results for Exercise 1](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/pathways/module2_lab/Mesen_vs_Immuno_edgeR.GseaPreranked.1465326037884.zip)
  
  [Exercise 2](http://bioinformatics-ca.github.io/PNAOD_module2_lab_gprofiler_2016/)
  
  [Precomputed results for exercise 2](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/pathways/module2_lab/gprofiler_results_mesenchymal.txt) 
  


[&uarr;](#back_to_top)

***

###  Module 3: Network Visualization and Analysis with Cytoscape <a id="module_3"></a>

  *<font color="#827e9c">Gary Bader</font>*
  
  [Lecture Part 1](https://bioinformatics.ca/pathways-module-3)
  
  [Lecture Part 2](https://bioinformatics.ca/pathways-module-3-part-2-2016)
  
#### Lab practical: Cytoscape Demo, Enrichment Map
  *<font color="#827e9c">Veronique Voisin</font>*
  
  [Exercise 1](http://bioinformatics-ca.github.io/PNAOD_module3_lab_EM_GSEA_2016/)
  
  [Exercise 2](http://bioinformatics-ca.github.io/PNAOD_module3_lab_EM_gprofiler_2016/)

[&uarr;](#back_to_top)

***

### Integrated Assignment <a id="assignment1"></a>

*<font color="#827e9c">Veronique Voisin</font>*

[Integrated Assignment: Enrichment Analysis Using GSEA and g:Profiler](http://bioinformatics-ca.github.io/PNAOD_integrated_assingment_1_2016/)

[Answers](https://bioinformatics.ca/pathways-integrated-assignment-1-answers-2016)

#### Data Processing

[Matrix](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/pathways/integrated_assignment_1/data_processing/GSE39491_series_matrix.txt)

[RScript](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/pathways/integrated_assignment_1/data_processing/code_integrated_assignment_BEvsNE.R)

[Metadata](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/pathways/integrated_assignment_1/data_processing/metadata.csv)

[&uarr;](#back_to_top)

***

##  Day 2 <a id="day_2"></a>

###  Module 4: More Depth on Pathway and Network Analysis <a id="module_4"></a>

  *<font color="#827e9c">Lincoln Stein</font>*
  
  [Lecture](https://bioinformatics.ca/pathways-module-4)
  
#### Lab practical: *De Novo* Subnetwork Clustering Analysis in Reactome
  
  *<font color="#827e9c">Robin Haw</font>*
  
  [Lecture](https://bioinformatics.ca/pathways-module-4-lab-slide-2016)
  
  [Exercise](http://bioinformatics-ca.github.io/PNAOD_module4_lab_2016/)
 
 [&uarr;](#back_to_top) 

***

###  Module 5: Gene Function Prediction  <a id="module_5"></a>

  *<font color="#827e9c">Quaid Morris</font>*
  
  [Lecture](https://bioinformatics.ca/pathways-module-5)
  
  Lab practical:
  
  [&uarr;](#back_to_top)

***

### Integrated Assignment <a id="assignment2"></a>

*<font color="#827e9c">Veronique Voisin</font>*

Integrated Assignment: Reactome FI and GeneMANIA

[&uarr;](#back_to_top)

***

##  Day 3 <a id="day_3"></a>

###  Module 6: Regulatory Network Analysis  <a id="module_6"></a>

  *<font color="#827e9c">Michael Hoffman</font>*
  
[Lecture](https://bioinformatics.ca/pathways-module-6)
  
#### Lab practical: iRegulon

Document:

Data:

Precomputed results:
To use the precomputed results: 1) launch Cytoscape ; 2) open the "prostate_cancer_genemania_network.cys" file ; 3) go to App > iRegulon > 'Load results from file...'


[&uarr;](#back_to_top)  

***
